### ISSUE TITLE
JRuby: Nokogiri::XML::XPath::SyntaxError when querying with pseudoclass

### ISSUE THREAD

**Issue by mattparlane** [sparklemotion/nokogiri#1470]
This is with JRuby 9.1.1.0, and Nokogiri 1.6.7.2:

``` ruby
require 'nokogiri'
p Nokogiri.XML('<foo></foo>').css('a:hover img')
```

MRI:

```
# /usr/local/bin/ruby.mri test.rb
[]
```

JRuby:

```
# ruby test.rb
Nokogiri::XML::XPath::SyntaxError: //a[hover(.)]//img
        evaluate at nokogiri/XmlXpathContext.java:169
  block in xpath at /root/jruby-9.1.1.0/lib/ruby/gems/shared/gems/nokogiri-1.6.7.2-java/lib/nokogiri/xml/searchable.rb:165
             map at org/jruby/RubyArray.java:2355
           xpath at /root/jruby-9.1.1.0/lib/ruby/gems/shared/gems/nokogiri-1.6.7.2-java/lib/nokogiri/xml/searchable.rb:156
    css_internal at /root/jruby-9.1.1.0/lib/ruby/gems/shared/gems/nokogiri-1.6.7.2-java/lib/nokogiri/xml/searchable.rb:193
             css at /root/jruby-9.1.1.0/lib/ruby/gems/shared/gems/nokogiri-1.6.7.2-java/lib/nokogiri/xml/searchable.rb:107
     <top> at test.rb:2
```

**Comment by flavorjones** [comment]
Hi, thanks for submitting this bug report. Certainly we should make both MRI and JRuby implementations behave the same.

**Comment by flavorjones** [comment]
This is fixed in v1.15.0, likely because of the work in #2867 and specifically this commit: 892b2c2d
